reactor-netty-core/src/main/java/reactor/netty/transport/AddressUtils.java

```diff
@@ -88,30 +88,46 @@ public static InetSocketAddress createUnresolved(String hostname, int port) {
 	}
 
 	/**
-	 * Parse unresolved InetSocketAddress. Numeric IP addresses will be detected and
-	 * resolved.
+	 * Parse unresolved InetSocketAddress. Numeric IP addresses will be detected and resolved.
 	 *
 	 * @param address ip-address or hostname
 	 * @param defaultPort the default port
-	 * @return InetSocketAddress for given parameters
+	 * @return {@link InetSocketAddress} for given parameters, only numeric IP addresses will be resolved
 	 */
 	public static InetSocketAddress parseAddress(String address, int defaultPort) {
+		return parseAddress(address, defaultPort, false);
+	}
+
+	/**
+	 * Parse unresolved InetSocketAddress. Numeric IP addresses will be detected and resolved.
+	 *
+	 * @param address ip-address or hostname
+	 * @param defaultPort the default port
+	 * @param strict if true throws an exception when the address cannot be parsed
```

Comment:
maybe mention clearly that if `false`, an _unresolved_ `INetSocketAddress` is instead returned, and that it can include the case of the host having been parsed but not the port (replaced by `defaultPort`)

Reply:
fixed